### PR TITLE
Attempt at sharing database object between server and library interfaces

### PR DESCRIPTION
**Pull request by ErikBjare**

### CODE REVIEW

File: src/android/mod.rs [hunk 1]
```diff
@@ -25,16 +25,19 @@ pub mod android {
...
     use datastore::Datastore;
     use models::{Event, Bucket};
+    use endpoints;
 
-    fn openDatastore() -> Datastore {
-        let db_dir = dirs::db_path().to_str().unwrap().to_string();
-        Datastore::new(db_dir)
-    }
+
+    static mut DATASTORE: Option<&Datastore> = None;
```
Comment (johan-bjareholt) [inline src/android/mod.rs]: I've seen this way of making Singletons with Options a few times before in a couple of gtk-rs projects, not a perfect solution but it's fine.

Comment (johan-bjareholt) [reply src/android/mod.rs]: Just realized though that those projects have them inside a atomic reference counter and a mutex which is not possible here.

Comment (johan-bjareholt) [reply src/android/mod.rs]: The best solution would probably be so DatabaseWorker is a singleton as the actual Database object is just a wrapper with a Requester from mpsc_requests which could be cloned without any issues.

**Comment by johan-bjareholt** [comment]
I now have a prototype where the Database object is clone-able.
Would this solve the issue for you?

**Comment by ErikBjare** [comment]
@johan-bjareholt That would work very nicely, yes!

**Comment by johan-bjareholt** [comment]
Here's the PR #20

**Comment by ErikBjare** [comment]
Closing, solved in #20

**Pull request closed without merging**